openiked/patches/41-direct_mode.patch

```diff
@@ -0,0 +1,56 @@
+diff -ruN a/iked/iked.c b/iked/iked.c
+--- a/iked/iked.c	2022-01-24 10:37:17.786149145 +0100
++++ b/iked/iked.c	2022-01-28 10:12:51.143080799 +0100
+@@ -111,7 +111,7 @@
+ {
+ 	const char *progname = "iked";
+ 
+-	fprintf(stderr, "usage: %s [-dnSTtv] [-D macro=value] "
++	fprintf(stderr, "usage: %s [-dnSTtvr] [-D macro=value] "
+ 	    "[-f file] [-p udpencap_port] [-s socket]\n", progname);
+ 	exit(1);
+ }
+@@ -120,7 +120,7 @@
+ main(int argc, char *argv[])
+ {
+ 	int		 c;
+-	int		 debug = 0, verbose = 0;
++	int		 debug = 0, verbose = 0, direct = 0;
+ 	int		 opts = 0;
+ 	enum natt_mode	 natt_mode = NATT_DEFAULT;
+ 	in_port_t	 port = IKED_NATT_PORT;
+@@ -147,7 +147,7 @@
+ 	argv = saved_av;
+ #endif
+ 
+-	while ((c = getopt(argc, argv, "6D:df:np:Ss:Ttv")) != -1) {
++	while ((c = getopt(argc, argv, "6D:df:np:Ss:Ttvr")) != -1) {
+ 		switch (c) {
+ 		case '6':
+ 			log_warnx("the -6 option is ignored and will be "
+@@ -196,6 +196,9 @@
+ 			verbose++;
+ 			opts |= IKED_OPT_VERBOSE;
+ 			break;
++		case 'r':
```

Comment:
what direct mode does?

I've tried executing  `/usr/bin/iked -r -vvv -f /etc/iked.conf` and most logs are still in syslog (apart from some hexdumps of packets?)

Reply:
Similar to -d but uses syslog.

Reply:
ok, I believe outputting hexes to stdout when using `-r -vvv` is expected